prov/gni/src/gnix_vc.c

```diff
@@ -682,10 +682,21 @@ static int __gnix_vc_connect_to_self(struct gnix_vc *vc)
 	vc->peer_irq_mem_hndl = ep->nic->irq_mem_hndl;
 
 	ret = _gnix_vc_schedule(vc);
-	if (ret != FI_SUCCESS)
+	if (ret == FI_SUCCESS) {
+		/*
+		 * give the TX queue a push
+		 */
+		ret = __gnix_vc_push_tx_reqs(vc);
```

Comment:
Why can't you put the warning in __gnix_vc_push_tx_reqs before returning?
Maybe you want to know what the calling function is?
Seems like a lot of code duplication.